application/src/main/java/org/eclipse/hono/application/ApplicationConfig.java

```diff
@@ -33,6 +33,17 @@
     private static final Vertx vertx = Vertx.vertx();
 
     /**
+     * Returns the HonoApplication instance responsible for initializing the
+     * Hono server components.
+     * 
+     * @return HonoApplication instance
+     */
+    @Bean
+    public HonoApplication honoApplication() {
```

Comment:
Well, I guess it's a matter of taste. Having the (possibly overridden) beans all defined in the ApplicationConfig or using inheritance in the Application class.
But yes, it's probably more concise to just let Application extend HonoApplication.